### PR TITLE
check brick type for volume creation

### PR DESCRIPTION
**Pull request by Madhu-1**
Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
Fixes: #1271 
Fixes: #1272

**Comment by Madhu-1** [comment]
@prashanthpai any issue with centos CI?

**Comment by Madhu-1** [comment]
@prashanthpai @aravindavk please review

**Comment by Madhu-1** [comment]
retest this please

File: glusterd2/commands/volumes/volume-create-txn.go [hunk 1]
```diff
@@ -92,6 +92,15 @@ func populateSubvols(volinfo *volume.Volinfo, req *api.VolCreateReq) error {
 			if subvolreq.ReplicaCount != 2 || subvolreq.ArbiterCount != 1 {
 				return errors.New("for arbiter configuration, replica count must be 2 and arbiter count must be 1. The 3rd brick of the replica will be the arbiter")
 			}
+			var abriterBrickCount uint16
+			for _, b := range subvolreq.Bricks {
+				if b.Type == "arbiter" {
+					abriterBrickCount++
+				}
+			}
+			if abriterBrickCount != 1 {
```
Comment (aravindavk) [inline glusterd2/commands/volumes/volume-create-txn.go]: You can change this to `if abriterBrickCount != subvolreq.ArbiterCount`. Currently only 1 is supported, in future if more than one arbiter bricks supported then only change required in Line 92

File: glusterd2/commands/volumes/volume-create-txn.go [hunk 2]
```diff
@@ -92,6 +92,15 @@ func populateSubvols(volinfo *volume.Volinfo, req *api.VolCreateReq) error {
 			if subvolreq.ReplicaCount != 2 || subvolreq.ArbiterCount != 1 {
 				return errors.New("for arbiter configuration, replica count must be 2 and arbiter count must be 1. The 3rd brick of the replica will be the arbiter")
 			}
+			var abriterBrickCount uint16
+			for _, b := range subvolreq.Bricks {
+				if b.Type == "arbiter" {
```
Comment (aravindavk) [inline glusterd2/commands/volumes/volume-create-txn.go]: `strings.Lower` is required here

Comment (vpandey-RH) [reply glusterd2/commands/volumes/volume-create-txn.go]: @Madhu-1 Can we add this checks and other if conditions reagrding replicaCount, disperseCount, etc, inside 
switch subvolreq.Type{} in the previous lines.

Comment (Madhu-1) [reply glusterd2/commands/volumes/volume-create-txn.go]: @vpandey-RH i think it's not required, @aravindavk  any comments?

**Comment by Madhu-1** [comment]
@aravindavk addressed review comments PTAL

File: glusterd2/commands/volumes/volume-create-txn.go [hunk 3]
```diff
@@ -92,6 +92,15 @@ func populateSubvols(volinfo *volume.Volinfo, req *api.VolCreateReq) error {
...
 				return errors.New("for arbiter configuration, replica count must be 2 and arbiter count must be 1. The 3rd brick of the replica will be the arbiter")
 			}
+			var abriterBrickCount int
+			for _, b := range subvolreq.Bricks {
+				if strings.ToLower(b.Type) == "arbiter" {
+					abriterBrickCount++
+				}
+			}
+			if abriterBrickCount != subvolreq.ArbiterCount {
+				return errors.New("for arbiter configuration one brick should be of type arbiter")
```
Comment (aravindavk) [inline glusterd2/commands/volumes/volume-create-txn.go]: message needs rephrase

File: glusterd2/commands/volumes/volume-create-txn.go [hunk 4]
```diff
@@ -92,6 +92,15 @@ func populateSubvols(volinfo *volume.Volinfo, req *api.VolCreateReq) error {
...
+			}
 			s.ArbiterCount = 1
```
Comment (aravindavk) [inline glusterd2/commands/volumes/volume-create-txn.go]: this also you can change `s.ArbiterCount = subvolreq.ArbiterCount`

**Comment by Madhu-1** [comment]
@aravindavk PTAL

File: glusterd2/volume/struct.go [hunk 5]
```diff
@@ -151,11 +152,13 @@ func NewBrickEntries(bricks []api.BrickReq, volName, volfileID string, volID uui
 			return nil, e
 		}
 
-		switch b.Type {
+		switch strings.ToLower(b.Type) {
```
Comment (prashanthpai) [inline glusterd2/volume/struct.go]: `strings.ToLower` shouldn't be needed. The request validation phase should catch and reject. The API shouldn't accept "Arbiter" but only "arbiter"

**Comment by Madhu-1** [comment]
@aravindavk @prashanthpai PTAL

**Pull request merged**